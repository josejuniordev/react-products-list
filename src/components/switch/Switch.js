import React, { useState } from 'react';
import './Switch.scss';
import PropTypes from 'prop-types';

function Switch(
  {
    checked = false,
    onChange = () => {},
  }
) {
  const [id] = useState(+new Date());
  const [checkedState, setCheckedState] = useState(checked);

  function onChangeHandler(event) {
    const isChecked = event.target.checked;
    onChange(isChecked);
    setCheckedState(isChecked);
  }

  return (
    <div className='switch-checkbox'>
      <input checked={checkedState} onChange={onChangeHandler} id={id} className='switch-checkbox__input' type="checkbox"/>
      <label htmlFor={id} className='switch-checkbox__switch'></label>
      <span className='switch-checkbox__label'>label do switch</span>
    </div>
  )
}

Switch.propTypes = {
  checked: PropTypes.bool,
};

export default Switch;
