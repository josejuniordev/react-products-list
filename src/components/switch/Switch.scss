@import "../../styles/bem";
@import "../../styles/colors";

.switch-checkbox {
  display: flex;
  align-items: center;
  position: relative;

  @include element(label) {
    font-size: 12px;
    line-height: 21px;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    color: $light-gray;
    cursor: pointer;
  }

  @include element(input) {
    width: 0;
    height: 0;
    position: absolute;
    overflow: hidden;

    $switch-spacing: 2px;
    $switch-height: 10px;
    $switch-width: 20px;

    + .switch-checkbox__switch {
      color: $white;
      cursor: pointer;
      height: $switch-height;
      background: $disabled-color;
      width: $switch-width;
      display: block;
      position: relative;
      border-radius: 20px;
      margin-right: 8px;
      -webkit-transition: all .2s;
      -moz-transition: all .2s;
      -ms-transition: all .2s;
      -o-transition: all .2s;

      &:after {
        $dimension: $switch-height - $switch-spacing * 2;
        content: '';
        background: currentColor;
        top: $switch-spacing;
        left: $switch-width - $switch-spacing - $dimension;
        width: $dimension;
        height: $dimension;
        -webkit-border-radius: 3px;
        -moz-border-radius: 3px;
        border-radius: 3px;
        position: absolute;
        -webkit-transition: all .2s;
        -moz-transition: all .2s;
        -ms-transition: all .2s;
        -o-transition: all .2s;
      }
    }

    &:checked + .switch-checkbox__switch {
      background: $blue;

      &:after {
        left: $switch-spacing;
      }
    }
  }
}
