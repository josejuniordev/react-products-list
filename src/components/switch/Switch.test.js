import { shallow } from 'enzyme';
import React from 'react';
import Switch from './Switch';

describe('<Switch />', () => {
  const rendered = shallow(<Switch />);

  it('should render correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
