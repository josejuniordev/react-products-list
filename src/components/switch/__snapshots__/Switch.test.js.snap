// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render correctly 1`] = `ShallowWrapper {}`;
